src/biogeochem/DustEmisFactory.F90

```diff
@@ -28,30 +28,28 @@ function create_dust_emissions(bounds, NLFilename) result(dust_emis)
         !---------------------------------------------------------------------------
         use DustEmisBase      , only : dust_emis_base_type
         use DustEmisZender2003, only : dust_emis_zender2003_type
-        use clm_varctl        , only : dust_emis_method
         use decompMod         , only : bounds_type
         use shr_kind_mod      , only : CL => shr_kind_cl
+        use shr_dust_emis_mod , only : is_dust_emis_zender, is_dust_emis_leung
```

Comment:
I suggest a short justification explaining why this is preferable over using the generic dust_emis_method variable.

Reply:
Yep will do.

Reply:
Adding some notes here. The design doc for the CMEPS code is here:
[
](https://docs.google.com/document/d/18nZ3LJF5W-YF9iBhqed6s_NWeKOvSSL2-k0Lye1nnLg)

It uses a OOP design in order to make sure that it can't be messed up by either CAM or CTSM (since it's used by both). Using the function calls is a way of ensuring internal variables aren't able to be changed by either. There's a few other ways it could be done, but since it's used by both I thought it important to make sure neither could mess with the internal settings. That's also the reason to have a unit tester for it, so that both models can rely on the behavior for it.

Reply:
@slevis-lmwg I added a note in this regard at the function header. Let me know about any feedback.

Reply:
Looks good. Thank you for doing that @ekluzek